src/workerd/api/tests/js-rpc-test.js

```diff
@@ -832,6 +860,118 @@ export let serializeRpcPromiseOrProprety = {
   },
 }
 
+export let streams = {
+  async test(controller, env, ctx) {
+    // Send WritableStream.
+    {
+      let { readable, writable } = new IdentityTransformStream();
+      let promise = env.MyService.writeToStream(writable);
+      let text = await new Response(readable).text();
+      assert.strictEqual(text, "foo, bar, baz!");
+      await promise;
+    }
+
+    // TODO(someday): Test JS-backed WritableStream, when it actually works.
+
+    // TODO(someday): Is there any way to construct an encoded WritableStream? Only system
+    //   streams can be encoded, but there's no API that returns an encoded WritableStream I think.
+
+    // Send ReadableStream.
+    {
+      let { readable, writable } = new IdentityTransformStream();
+      let promise = env.MyService.readFromStream(readable);
+
+      let writer = writable.getWriter();
+      let enc = new TextEncoder();
+      await writer.write(enc.encode("foo, "));
+      await writer.write(enc.encode("bar, "));
+      await writer.write(enc.encode("baz!"));
+      await writer.close();
+
+      assert.strictEqual(await promise, "foo, bar, baz!");
+    }
+
+    // Send a JS-backed ReadableStream.
+    {
+      let controller;
+      let readable = new ReadableStream({
+        start(c) { controller = c; }
+      });
+      let promise = env.MyService.readFromStream(readable);
+
+      let enc = new TextEncoder();
+      controller.enqueue(enc.encode("foo, "));
+      controller.enqueue(enc.encode("bar, "));
+      controller.enqueue(enc.encode("baz!"));
+      controller.close();
+
+      assert.strictEqual(await promise, "foo, bar, baz!");
+    }
+
+    // Send a ReadableStream that is locked.
+    {
+      let { readable, writable } = new IdentityTransformStream();
+
+      let reader = readable.getReader();
+
+      assert.rejects(env.MyService.readFromStream(readable), {
+        name: "TypeError",
+        message: "The ReadableStream has been locked to a reader."
+      });
```

Comment:
Might be good to perform a `reader.read()` after this check to ensure that the state of the stream has not changed.

Reply:
Also... similar test for `writable`?

Reply:
39ae6632